assets/src/js/dokan-product-category-ui.js

```diff
@@ -0,0 +1,379 @@
+
+            categoriesState = newCategories;
+            ProductCategory.updateCategorySelection( catlevel, termId );
+        },
+
+        scrollTo: ( to = 0 ) => {
+            $( '#dokan-single-categories' ).animate( { scrollLeft: 300 * to }, 800 );
+        },
+
+        indicatorScrollTo: ( left = true ) => {
+            $( '#dokan-single-categories' ).animate( { scrollLeft: `${ left ? '+' : '-' }=350px` }, 800 );
+        },
+
+        setCatId: ( id ) => {
+            let ui = `<input type="hidden" name="${ dokan_product_category_data.is_single ? 'product_cat' : 'product_cat[]' }" class="dokan_product_cat" id="dokan_product_cat" value="${ id }"></input>`;
+            ui += `<input type="hidden" class="dokan_chosen_product_cat" name="chosen_product_cat[]" value="${ id }"></input>`;
+
+            let category = dokan_product_category_data.categories.filter( ( element, index ) => {
+                return element.cat_ID == id;
+            });
+
+            if ( ! dokan_product_category_data.is_single ) {
+                ui += category[0].parents.map( ( element, index ) => {
+                    return `<input type="hidden" name="product_cat[]" class="dokan_product_cat" id="dokan_product_cat" value="${ element.cat_ID }"></input>`;
+                }).join('');
+            }
+
+            $( `.dokan-cih-level-${ inputHolder }` ).html( ui );
+        },
+
+        setCatName: ( name ) => {
+            $( `.dokan-ssct-level-${ inputHolder }` ).html( name );
+        },
+
+        addANewCatBox: () => {
+            boxCounter++;
+            let html = `
+                <div class="dokan-select-product-category-container">
+                    <div class="dokan-form-group dokan-select-product-category" data-dokansclevel="${ boxCounter }" id="dokan-category-open-modal">
+                        <span id="dokan_product_cat_res" class="dokan-select-product-category-title dokan-ssct-level-${ boxCounter }">- Select a category -</span>
+                        <span class="dokan-select-product-category-icon"><i class="fas fa-edit"></i></span>
+                    </div>
+                        ${ ! dokan_product_category_data.is_single ? `
+                        <div class="dokan-select-product-category-remove-container">
+                            <span class="dokan-select-product-category-remove"><i class="fas fa-times"></i></span>
+                        </div>`
+                        : ''}
+                    <span class="dokan-cat-inputs-holder dokan-cih-level-${ boxCounter }" ></span>
+                </div>
+                `;
+
+            $( '.dokan-add-new-cat-box' ).append( html );
+        },
+
+        findCategory: ( id ) => {
+            let allCategories = [ ...dokan_product_category_data.categories ];
+
+            return allCategories.findIndex( ( category, index ) => {
+                return id == category.cat_ID;
+            });
+        }
+
+    };
+
+    // On DOM ready.
+    $( function() {
```

Comment:
no need for another $( function(), rewrite this as below
```
$(document).ready(function(){
      ProductCategory.init();
});
````
after that move below code before .ready, eg:
```
$('body').on( 'click', '#dokan-category-open-modal', this.openModal );
```
where openModal function will  contain required code